src/components/SearchList.js

```diff
@@ -0,0 +1,45 @@
+import React, { useRef, useState } from 'react';
+import { normalizeValue } from './Helper';
+
+export default function SearchList(listItems) {
+  const [searchedItem, setSearchedItem] = useState('');
+  const searchInputRef = useRef();
+  //function to update list of items as user enters characters
+  const updateSearch = () => {
+    setSearchedItem(searchInputRef.current.value);
+  };
+
+  const clearSearch = () => {
+    setSearchedItem('');
+    searchInputRef.current.value = '';
+  };
+
+  //function to return list of items whose normalized names contain characters entered by the user
+
+  const filteredList = listItems.list.filter((item) => {
+    const normalizedSearch = normalizeValue(searchedItem);
+    return item.itemNameNormalize.includes(normalizedSearch);
+  });
+
+  return (
+    <div>
+      Filter Items:
+      <input
+        type="text"
+        ref={searchInputRef}
+        placeholder="Search for Items here"
+        onChange={updateSearch}
+      />
+      {searchedItem && <button onClick={clearSearch}>X</button>}
```

Comment:
Accessibility note: This button will need to have an `aria-label` added since the only value here is "X" it gives no information to a screen reader user.

| without aria-label  | with aria-label|
| ------------- | ------------- |
| <video width=400px src="https://user-images.githubusercontent.com/19564687/140624893-3bdfa8d4-bd86-4f4b-a591-9c81d33d0664.mov  " /> | <video width=400px src="https://user-images.githubusercontent.com/19564687/140624926-fa409237-a2b8-4b29-9ce9-90f47a86cf15.mov" /> |